### ISSUE TITLE
`InvalidOperationException` after an Entity is destroyed

### ISSUE THREAD

**Issue by sNaticY** [quabug/EntitiesBT#140]
This issue only happens when two or more entities have the same tree source. Any cleanup steps are required before destroying an Entity with BehaviourTreeRoot?

The full call stack is as follows.

```
InvalidOperationException: The BlobAssetReference is not valid. Likely it has already been unloaded or released.
Unity.Entities.BlobAssetReferenceData.ValidateNonBurst () (at Library/PackageCache/com.unity.entities@0.16.0-preview.21/Unity.Entities/Blobs.cs:253)
Unity.Entities.BlobAssetReferenceData.ValidateNotNull () (at Library/PackageCache/com.unity.entities@0.16.0-preview.21/Unity.Entities/Blobs.cs:270)
Unity.Entities.BlobAssetReference`1[T].get_Value () (at Library/PackageCache/com.unity.entities@0.16.0-preview.21/Unity.Entities/Blobs.cs:358)
EntitiesBT.Entities.NodeBlobRef.get__blob () (at Library/PackageCache/com.quabug.entities-bt.essential@0.21.10/Runtime/Entities/NodeBlob.cs:43)
EntitiesBT.Entities.NodeBlobRef.GetTypeId (System.Int32 nodeIndex) (at Library/PackageCache/com.quabug.entities-bt.essential@0.21.10/Runtime/Entities/NodeBlob.cs:49)
EntitiesBT.Core.VirtualMachine.Tick[TNodeBlob,TBlackboard] (System.Int32 index, TNodeBlob& blob, TBlackboard& bb) (at Library/PackageCache/com.quabug.entities-bt.essential@0.21.10/Runtime/Core/VirtualMachine.cs:22)
EntitiesBT.Core.VirtualMachine.Tick[TNodeBlob,TBlackboard] (TNodeBlob& blob, TBlackboard& bb) (at Library/PackageCache/com.quabug.entities-bt.essential@0.21.10/Runtime/Core/VirtualMachine.cs:15)
EntitiesBT.Entities.VirtualMachineSystem+<>c__DisplayClass_OnUpdate_LambdaJob0.OriginalLambdaBody (Unity.Entities.Entity entity, Unity.Entities.DynamicBuffer`1[T] buffers, EntitiesBT.Entities.CurrentBehaviorTreeComponent& currentBehaviorTree) (at Library/PackageCache/com.quabug.entities-bt.essential@0.21.10/Runtime/Entities/VirtualMachineSystem.cs:47)
EntitiesBT.Entities.VirtualMachineSystem+<>c__DisplayClass_OnUpdate_LambdaJob0.IterateEntities (Unity.Entities.ArchetypeChunk& chunk, EntitiesBT.Entities.VirtualMachineSystem+<>c__DisplayClass_OnUpdate_LambdaJob0+LambdaParameterValueProviders+Runtimes& runtimes) (at <cb18d41fc64e4dec8d7325227d4e62b0>:0)
EntitiesBT.Entities.VirtualMachineSystem+<>c__DisplayClass_OnUpdate_LambdaJob0.Execute (Unity.Entities.ArchetypeChunk chunk, System.Int32 chunkIndex, System.Int32 firstEntityIndex) (at <cb18d41fc64e4dec8d7325227d4e62b0>:0)
Unity.Entities.JobChunkExtensions.RunWithoutJobs[T] (T& jobData, Unity.Entities.ArchetypeChunkIterator& chunkIterator) (at Library/PackageCache/com.unity.entities@0.16.0-preview.21/Unity.Entities/IJobChunk.cs:182)
EntitiesBT.Entities.VirtualMachineSystem+<>c__DisplayClass_OnUpdate_LambdaJob0.RunWithoutJobSystem (Unity.Entities.ArchetypeChunkIterator* archetypeChunkIterator, System.Void* jobData) (at <cb18d41fc64e4dec8d7325227d4e62b0>:0)
Unity.Entities.InternalCompilerInterface.RunJobChunk[T] (T& jobData, Unity.Entities.EntityQuery query, Unity.Entities.InternalCompilerInterface+JobChunkRunWithoutJobSystemDelegate functionPointer) (at Library/PackageCache/com.unity.entities@0.16.0-preview.21/Unity.Entities/CodeGeneratedJobForEach/LambdaJobDescription.cs:378)
EntitiesBT.Entities.VirtualMachineSystem.OnUpdate () (at Library/PackageCache/com.quabug.entities-bt.essential@0.21.10/Runtime/Entities/VirtualMachineSystem.cs:32)
Unity.Entities.SystemBase.Update () (at Library/PackageCache/com.unity.entities@0.16.0-preview.21/Unity.Entities/SystemBase.cs:413)
Unity.Entities.ComponentSystemGroup.UpdateAllSystems () (at Library/PackageCache/com.unity.entities@0.16.0-preview.21/Unity.Entities/ComponentSystemGroup.cs:435)
UnityEngine.Debug:LogException(Exception)
Unity.Debug:LogException(Exception) (at Library/PackageCache/com.unity.entities@0.16.0-preview.21/Unity.Entities/Stubs/Unity/Debug.cs:19)
Unity.Entities.ComponentSystemGroup:UpdateAllSystems() (at Library/PackageCache/com.unity.entities@0.16.0-preview.21/Unity.Entities/ComponentSystemGroup.cs:440)
Unity.Entities.ComponentSystemGroup:OnUpdate() (at Library/PackageCache/com.unity.entities@0.16.0-preview.21/Unity.Entities/ComponentSystemGroup.cs:387)
Unity.Entities.ComponentSystem:Update() (at Library/PackageCache/com.unity.entities@0.16.0-preview.21/Unity.Entities/ComponentSystem.cs:113)
Unity.Entities.DummyDelegateWrapper:TriggerUpdate() (at Library/PackageCache/com.unity.entities@0.16.0-preview.21/Unity.Entities/ScriptBehaviourUpdateOrder.cs:333)
```

**Comment by quabug** [comment]
It is strange since `BehaviourTreeRoot` will create distinct `NodeBlob` of behavior tree, which means entity has its own runtime behavior tree and destroy an entity won't effect others.
Could you provide more details of this bug? Like reproduce steps or even an unity project? Thanks.

**Comment by sNaticY** [comment]
Thanks for your reply!

Here is a Minimal Sample Project to reproduce the issue [entities-bt-issue](https://github.com/sNaticY/entities-bt-issue)
Unity Version: 2020.1.7f1

**Comment by quabug** [comment]
The problem is entity reusing same `NodeBlob` while instantiate from `EntityManager`.
Turns out I have never tested instantiate an entity from code.
Thank you for reporting this bug.
It has been fixed in the latest update ([essential/v0.21.11](https://openupm.com/packages/com.quabug.entities-bt.essential/)).

**Comment by sNaticY** [comment]
Thank you!
